purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/properties/CornerRadiuses.kt

```diff
@@ -1,56 +1,128 @@
+            topTrailing: Double = this.topTrailing,
+            bottomLeading: Double = this.bottomLeading,
+            bottomTrailing: Double = this.bottomTrailing,
+        ): Dp {
+            return Dp(topLeading, topTrailing, bottomLeading, bottomTrailing)
+        }
+    }
+
     /**
-     * The bottom-trailing, or bottom-end, corner radius, in dp.
+     * Contains radius values for 4 corners, in percentages.
      */
-    @get:JvmSynthetic
-    @SerialName("bottom_trailing")
-    val bottomTrailing: Double,
-) {
-    companion object {
-        @get:JvmSynthetic val zero = CornerRadiuses(0.0, 0.0, 0.0, 0.0)
-
-        @get:JvmSynthetic val default = zero
+    @Poko
+    @Serializable
+    class Percentage(
+        /**
+         * The top-leading, or top-start, corner radius, in percentage [0-100].
+         */
+        @get:JvmSynthetic
+        @IntRange(from = 0, to = 100)
+        @SerialName("top_leading")
+        val topLeading: Int,
+        /**
+         * The top-trailing, or top-end, corner radius, in percentage [0-100].
+         */
+        @get:JvmSynthetic
+        @IntRange(from = 0, to = 100)
+        @SerialName("top_trailing")
+        val topTrailing: Int,
+        /**
+         * The bottom-leading, or bottom-start, corner radius, in percentage [0-100].
+         */
+        @get:JvmSynthetic
+        @IntRange(from = 0, to = 100)
+        @SerialName("bottom_leading")
+        val bottomLeading: Int,
+        /**
+         * The bottom-trailing, or bottom-end, corner radius, in percentage [0-100].
+         */
+        @get:JvmSynthetic
+        @IntRange(from = 0, to = 100)
+        @SerialName("bottom_trailing")
+        val bottomTrailing: Int,
+    ) : CornerRadiuses {
+
+        constructor(all: Int) : this(all, all, all, all)
     }
+}
 
-    constructor(all: Double) : this(all, all, all, all)
+@OptIn(InternalRevenueCatAPI::class)
+internal object CornerRadiusesSerializer : KSerializer<CornerRadiuses> {
+    private val serializer = CornerRadiuses.Dp.serializer()
+    override val descriptor: SerialDescriptor = serializer.descriptor
 
-    fun copy(
-        topLeading: Double = this.topLeading,
-        topTrailing: Double = this.topTrailing,
-        bottomLeading: Double = this.bottomLeading,
-        bottomTrailing: Double = this.bottomTrailing,
-    ): CornerRadiuses {
-        return CornerRadiuses(topLeading, topTrailing, bottomLeading, bottomTrailing)
+    override fun serialize(encoder: Encoder, value: CornerRadiuses) {
+        // We don't need to serialize CornerRadiuses.
+    }
+
+    /**
+     * For now, the backend will always return corner radiuses in Dp, so this will assume all serializations
+     * are done with that serializer.
+     */
+    override fun deserialize(decoder: Decoder): CornerRadiuses {
+        return try {
+            decoder.decodeSerializableValue(serializer)
+        } catch (@Suppress("TooGenericExceptionCaught") e: Throwable) {
```

Comment:
I don't fully like this... But `ShapeTests` and `MaskShapeTests` were failing to deserialize for me when corner radiuses were provided as "null"...